Alignment/CommonAlignment/plugins/FilterOutLowPt.cc

```diff
@@ -0,0 +1,183 @@
+  runControl_ = iConfig.getUntrackedParameter<bool>("runControl",false);
+  runControlNumbers_ = iConfig.getUntrackedParameter<std::vector<unsigned int> >("runControlNumber",defaultRuns);
+
+  edm::InputTag TrackCollectionTag_ = iConfig.getUntrackedParameter<edm::InputTag>("src",edm::InputTag("generalTracks"));
+  theTrackCollectionToken = consumes<reco::TrackCollection>(TrackCollectionTag_);
+  
+}
+
+FilterOutLowPt::~FilterOutLowPt()
+{
+}
+
+void FilterOutLowPt::beginJob(){
+  trials=0;
+  passes=0;
+}
+
+bool FilterOutLowPt::filter( edm::Event& iEvent, const edm::EventSetup& iSetup)
+{
+
+  bool passesRunControl = false;
+  
+  if(runControl_){
+    if (debugOn){
+      for(unsigned int i=0;i<runControlNumbers_.size();i++){
+	edm::LogInfo("FilterOutLowPt")<<"run number:" <<iEvent.id().run()<<" keeping runs:"<<runControlNumbers_[i]<<std::endl;
+      }
+    }
+
+    for(unsigned int j=0;j<runControlNumbers_.size();j++){
+      if(iEvent.eventAuxiliary().run() == runControlNumbers_[j]){ 
+	if (debugOn){
+	  edm::LogInfo("FilterOutLowPt")<<"run number"<< runControlNumbers_[j] << " match!"<<std::endl;
+	}
+	passesRunControl = true;
+	break;
+      }
+    }
+    if (!passesRunControl) return false;
+  }
+  
+  trials++;
+
+  bool accepted = false;
+  float fraction = 0;  
+  // get GeneralTracks collection
+
+  edm::Handle<reco::TrackCollection> tkRef;
+  iEvent.getByToken(theTrackCollectionToken,tkRef);    
+  const reco::TrackCollection* tkColl = tkRef.product();
+ 
+  int numhighpurity=0;
+  _trackQuality = reco::TrackBase::qualityByName("highPurity");
+
+  if(tkColl->size()>numtrack){ 
+    reco::TrackCollection::const_iterator itk = tkColl->begin();
+    reco::TrackCollection::const_iterator itk_e = tkColl->end();
+    for(;itk!=itk_e;++itk){
+      if( itk->quality(_trackQuality) &&
+	  (itk->pt() >= ptmin) 
+	  ) numhighpurity++;
+    }
+    fraction = numhighpurity; //(float)tkColl->size();
+    if(fraction>=thresh) accepted=true;
+  }
+    
+  if (debugOn) {
+    int ievt = iEvent.id().event();
+    int irun = iEvent.id().run();
+    int ils  = iEvent.luminosityBlock();
+    int bx   = iEvent.bunchCrossing();
+      
+    edm::LogInfo("FilterOutLowPt")<<" Run " << irun << " Event " << ievt << " Lumi Block " << ils << " Bunch Crossing " << bx << " Fraction " << fraction << " NTracks " << tkColl->size() << " Accepted " << accepted << std::endl;
+
+  }
+ 
+  // count the trials and passes
+  unsigned int iRun = iEvent.id().run(); 
+  if (eventsInRun_.count(iRun)>0){
+    eventsInRun_[iRun].first+=1;
+    if(accepted) eventsInRun_[iRun].second+=1;
+  } else {
+    std::pair<int,int> mypass = std::make_pair(1,0);
+    if(accepted) mypass.second = 1;
+    eventsInRun_[iRun]= mypass;
+  }
+
+  if (applyfilter){
+    if(accepted) passes++;
+    return accepted;
+  } else
+    return true;
+  
+}
+
+void FilterOutLowPt::endJob(){
+  
+  double eff =  passes/trials;
+  double eff_err = TMath::Sqrt((eff*(1-eff))/trials);
```

Comment:
@mmusich I believe, it is recommended to use `std::sqrt` instead of the ROOT functions